program/tests/stake_instruction.rs

```diff
@@ -0,0 +1,7076 @@
+#![allow(clippy::arithmetic_side_effects)]
+
+use {
+    assert_matches::assert_matches,
+    bincode::serialize,
+    mollusk_svm::{result::Check, Mollusk},
+    solana_account::{AccountSharedData, ReadableAccount, WritableAccount},
+    solana_program_runtime::loaded_programs::ProgramCacheEntryOwner,
+    solana_sdk::{
+        account::create_account_shared_data_for_test,
+        account_utils::StateMut,
+        feature_set::stake_raise_minimum_delegation_to_1_sol,
+        instruction::{AccountMeta, Instruction},
+        program_error::ProgramError,
+        pubkey::Pubkey,
+        stake::{
+            config as stake_config,
+            instruction::{
+                self, authorize_checked, authorize_checked_with_seed, initialize_checked,
+                set_lockup_checked, AuthorizeCheckedWithSeedArgs, AuthorizeWithSeedArgs,
+                LockupArgs, StakeError, StakeInstruction,
+            },
+            stake_flags::StakeFlags,
+            state::{
+                warmup_cooldown_rate, Authorized, Delegation, Lockup, Meta, Stake, StakeAuthorize,
+                StakeStateV2,
+            },
+            MINIMUM_DELINQUENT_EPOCHS_FOR_DEACTIVATION,
+        },
+        stake_history::{Epoch, StakeHistoryEntry},
+        system_program,
+        sysvar::{
+            clock::{self, Clock},
+            epoch_rewards::{self, EpochRewards},
+            epoch_schedule::{self, EpochSchedule},
+            rent::{self, Rent},
+            rewards,
+            stake_history::{self, StakeHistory},
+        },
+        transaction_context::TransactionReturnData,
+    },
+    solana_stake_program::{get_minimum_delegation, id},
+    solana_vote_program::{
+        self,
+        vote_state::{self, VoteState, VoteStateVersions},
+    },
+    std::{collections::HashSet, str::FromStr},
+    test_case::test_case,
+};
+
+fn mollusk_native() -> Mollusk {
+    let mut mollusk = Mollusk::default();
+    mollusk
+        .feature_set
+        .deactivate(&stake_raise_minimum_delegation_to_1_sol::id());
+    mollusk
+}
+
+fn mollusk_bpf() -> Mollusk {
+    let mut mollusk = Mollusk::new(&id(), "solana_stake_program");
+    mollusk
+        .feature_set
+        .deactivate(&stake_raise_minimum_delegation_to_1_sol::id());
+    mollusk
+}
```

Comment:
Sorry! I need to update the documentation, but you can actually use one Mollusk instance if you wanted. It doesn't map to a program anymore. `new` is just a convenience method to create a new instance and load a BPF program into the cache, but a Mollusk instance comes with a loaded cache and can be pointed at any program in the cache, doesn't matter if you use `default` or `new`.

The program to invoke is dictated by the instruction's program ID.

Reply:
i believe what im doing here is correct (note that this branch uses a local version of mollusk on my machine that includes all three of my open prs). `mollusk_native()` loads no program so it uses the native stake program, `mollusk_bpf()` loads the bpf stake program which overrides the native. they have the same program id so there isnt a way to discriminate between them otherwise

Reply:
Ahh I see, so the main purpose being to overwrite the stake program in the second one. Yeah, I think you've got it right then.